pkg/browser/browser.go

```diff
@@ -0,0 +1,80 @@
+// Package browser facilitates opening of URLs in a web browser.
+package browser
+
+import (
+	"io"
+	"os"
+	"os/exec"
+
+	cliBrowser "github.com/cli/browser"
+	"github.com/cli/go-gh/pkg/config"
+	"github.com/cli/safeexec"
+	"github.com/google/shlex"
+)
+
+// Browser represents a web browser that can be used to open up URLs.
+type Browser struct {
+	launcher string
+	stderr   io.Writer
+	stdout   io.Writer
+}
+
+// NewBrowser initializes a Browser. If a launcher is not specified
+// one is determined based on environment variables or from the
+// configuration file.
+// The order of precedence for determining a launcher is:
+// - Specified launcher;
+// - GH_BROWSER environment variable;
+// - browser option from configuration file;
+// - BROWSER environment variable.
+func NewBrowser(launcher string, stdout, stderr io.Writer) Browser {
```

Comment:
Since we are in the `browser` package, is it redundant to call the initializer function `NewBrowser`, or would it suffice to just have `New`? I'm fine either way; I'm just wondering about our stance on naming across micro-packages.

Reply:
Agreed, good call. I like `New` better.